app/views/events/show.html.erb

```diff
@@ -1,46 +1,68 @@
-<%- model_class = Event -%>
+<%- model_class = @event.class -%>
 <div class="page-header">
   <h1>
-    <%=t '.title', :default => model_class.model_name.human.titleize %>
-    <% if user_signed_in? %>
-      <span class="pull-right">
-        <% if current_user.organizing_events.include?(@event) %>
-          <% if can? :edit, @event %>
-              <%= link_to t('.edit', :default => t("helpers.links.edit")),
-                          edit_event_path(@event), :class => 'btn btn-default' %>
-          <% end %>
-          <% if can? :destroy, @event %>
-              <%= link_to t('.destroy', :default => t("helpers.links.destroy")),
-                          event_path(@event),
-                          :method => 'delete',
-                          :data => {:confirm => t('.confirm', :default => t("helpers.links.confirm"))},
-                          :class => 'btn btn-danger' %>
-          <% end %>
-        <% end %>
-      </span>
-    <% end %>
+    <%=t '.title', default: @event.model_name.human.titleize %>
   </h1>
+  <% if user_signed_in? %>
+    <span class="pull-right">
+      <% if current_user.organizing_events.include?(@event) %>
+        <% if can? :edit, @event %>
+          <%= link_to t('.edit', default: t("helpers.links.edit")),
+                      edit_event_path(@event), class: 'btn btn-default' %>
+        <% end %>
+        <% if can? :destroy, @event %>
+          <%= link_to t('.destroy', default: t("helpers.links.destroy")),
+                      event_path(@event),
+                      method: 'delete',
+                      data: {confirm: t('.confirm', default: t("helpers.links.confirm"))},
+                      class: 'btn btn-danger' %>
+        <% end %>
+      <% end %>
+    </span>
+  <% end %>
 </div>
 
 <dl class="dl-horizontal">
-  <% [:name, :description, :discipline, :game_mode, :max_teams, :deadline, :startdate, :enddate].each do |attribute| %>
-      <dt><strong><%= model_class.human_attribute_name(attribute) %>:</strong></dt>
-      <dd><%= @event.send attribute %></dd>
-  <% end %>
+  <dt><strong><%= model_class.human_attribute_name(:name) %>:</strong></dt>
```

Comment:
That really looks bad. Please do something about that duplication

Reply:
there is no nice way to do this. Views inherently have some amount of duplication, just look at the form fields for every model. The previous solution was arguably wrong, since it used `send` and couldn't handle injecting translations. I would ask you to provide a better solution.

Reply:
Just define a function that takes two parameters: the attribute (`:name`) and the value (`@event.name`) and call it 9 times

Reply:
I'm not sure if you can just define a method like that in the respective helper...
Otherwise, it would still look ugly:
```
{
  name: @event.name,
  discipline: @event.discipline,
  ...
  game_mode: @event.human_game_mode,
  ...
}
```

I would leave it as it is, since other loops depend on the fact that there will only be this format forever. This is the first time the previous pattern was broken (and i think it's not the last time).

Reply:
Maybe you are right. Keep it like that if you like